assets/src/edit-story/components/panels/design/preset/presetGroup.js

```diff
@@ -114,12 +120,26 @@ function PresetGroup({
     <Group ref={groupRef} type={type}>
       {presets.map((preset, i) => (
         <ButtonWrapper key={JSON.stringify(preset)} type={type}>
-          {itemRenderer(preset, i, activeIndex, handleClick, isEditMode)}
+          {itemRenderer(
+            preset,
+            i,
+            activeIndex,
+            handleClick,
+            isEditMode,
+            isLocal
+          )}
```

Comment:
This feels like too many positional arguments. Could we maybe invert the flow here using `{children}`, use a context, or at least use an object of arguments? The `itemRenderer` prop is drilled quite a bit, and I'm not a big fan.

Reply:
Yep, agreed, making changes here I also felt like this is a bit too much already and perhaps the `itemRenderer` as-is isn't really a good way to do it anymore. Now that the style and color panels differ more than before, perhaps it makes sense to take a fresh look at how similar things are shared between them at all -- perhaps it makes sense to split it up a bit more.

I'll do some refactoring within #3634 since that will need to use the saved colors, too.